map_model/src/map.rs

```diff
@@ -320,39 +320,35 @@ impl Map {
     /// The turns may belong to two different intersections!
     pub fn get_turns_from_lane(&self, l: LaneID) -> Vec<&Turn> {
         let lane = self.get_l(l);
-        let mut turns: Vec<&Turn> = self
+        let turns: Vec<&Turn> = self
             .get_i(lane.dst_i)
             .turns
             .iter()
-            .filter(|t| t.id.src == l)
+            .chain(
+                self.get_i(lane.src_i)
+                    .turns
+                    .iter()
+                    .take_while(|_| lane.is_walkable()),
```

Comment:
This relies on all of the walking turns being placed before the vehicle turns, right? This could be brittle; `filter` might be safer. If we do want the short-circuiting here, then we should add a docstring on `turns: Vec<Turn>` then

Reply:
No, this is the way I could find to `chain` all the turns from the source (i.e. contraflow) intersection only if `lane.is_walkable()`